### PR TITLE
Blacklist pillow 7.1.*

### PR DESCRIPTION
**Pull request by phue**
There is a regression in the latest Pillow releases

https://github.com/python-pillow/Pillow/pull/4528
The fix has been merged but not released yet

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by phue** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ requirements:
     - scipy >=0.19
     - matplotlib-base >=2.0.0
     - networkx >=2.0
-    - pillow >=4.3.0
+    - pillow >=4.3.0,!=7.1.*
```
Comment (soupault) [inline recipe/meta.yaml:49]: Should it be?:
```suggestion
    - pillow >=4.3.0,!=7.1.0,!=7.1.1
```

Comment (phue) [reply recipe/meta.yaml:49]: Assuming that the next point release has the fix, yes that's better

**Comment by jni** [comment]
Thanks for handling this, @phue!

I must say I don't understand the CI failures... Two appear to be some numerical precision issues that for some reason don't appear in the other builds, and the last one appears to be a dependency conflict.

The numerical precision issues could be easily dealt with with a patch, but I don't know how to deal with the dependency conflicts. @jakirkham? 😬

**Comment by phue** [comment]
@conda-forge-admin, please rerender

**Comment by phue** [comment]
> Thanks for handling this, @phue!
> 
> I must say I don't understand the CI failures... Two appear to be some numerical precision issues that for some reason don't appear in the other builds, and the last one appears to be a dependency conflict.
> 
> The numerical precision issues could be easily dealt with with a patch, but I don't know how to deal with the dependency conflicts. @jakirkham? grimacing

I don't understand them either.
Rerendering with latest `conda-smithy` made them disappear again. But now there is another error which I also don't understand, this time related to `tcl`

**Comment by hmaarrfk** [comment]
I'm not sure if this is going to have the effect you want.

THese kinds of issues are rather typical in rolling releases like conda-forge.

Maybe you can ask the package to be removed if the issue is serious enough?

**Comment by jakirkham** [comment]
I'm think we should just do a repodata patch if the issue is this version of `pillow` has issues. We could then apply this to both the current `scikit-image` package and older versions. WDYT?

**Comment by hmaarrfk** [comment]
> repodata patch 

I'm not sure what this is.

**Comment by jakirkham** [comment]
The idea would be to patch repodata for the `conda-forge` channel to alter the requirements of `scikit-image` packages. Patching is applied based on [code here]( https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/master/recipe/gen_patch_json.py ). Here's a somewhat recent patch that `distributed` needed as an example ( https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/32 ).

**Comment by hmaarrfk** [comment]
that sounds harder than simply pulling pillow 7.1

**Comment by jakirkham** [comment]
What about applying the upstream fix as a patch to `pillow` then?

**Comment by phue** [comment]
> I'm not sure if this is going to have the effect you want.
> 
> THese kinds of issues are rather typical in rolling releases like conda-forge.


Could you elaborate more on this? I currently don't understand why blacklisting is not a viable option. And I don't see how the CI failure is related to the change I made

**Comment by hmaarrfk** [comment]
> I currently don't understand why blacklisting is not a viable option.

The solver has been changing recently, so if somebody types `conda install pillow` then types `conda install scikit-image` I think they will get pillow 7.1.

> And I don't see how the CI failure is related to the change I made

Neither do I. I just skipped the test.

What platform do you use? I could just merge this even if it is failing if you want....

File: recipe/meta.yaml [hunk 2]
```diff
@@ -66,7 +66,11 @@ test:
   commands:
     - setx MPLBACKEND "Agg"  # [win]
     - export MPLBACKEND=Agg  # [unix]
-    - pytest --pyargs skimage
+    # I'm really not sure why why this test is complaining about tk
```
Comment (hmaarrfk) [inline recipe/meta.yaml:69]:
```suggestion
    # hmaarrfk: 2020/04/24
    # I'm really not sure why why this test is complaining about tk
```

**Comment by hmaarrfk** [comment]
honestly, i'm not too sure how the solver works out, so i'm willing to try this out, I just wanted to say that typically, the best way is to ping upstream for a fix, or to patch up the offending repo.

**Comment by phue** [comment]
> The solver has been changing recently, so if somebody types `conda install pillow` then types `conda install scikit-image` I think they will get pillow 7.1.

OK, now I get it that's unfortunate. I think the current situation is even worse though, because anyone typing `conda install scikit-image` will get a dysfunctional installation

> honestly, i'm not too sure how the solver works out, so i'm willing to try this out, I just wanted to say that typically, the best way is to ping upstream for a fix, or to patch up the offending repo.

I fully agree on that, in this case the fix has already been merged upstream so it's just pending a new release. Patching the pillow feedstock is of course an option, I just thought this would be in line with what was done for pip [here](https://github.com/scikit-image/scikit-image/commit/e094b3829d74cff4beb301770ee79545a4c53d1a)

**Comment by hmaarrfk** [comment]
it seems that you are hitting a nasty bug on OSX.. even if i merged, it would give you inconstient result on different platforms

**Comment by hmaarrfk** [comment]
i'm ok skipping those tk tests, but not so sure about skipping the OSX test.

I'll let @jni review before merging

**Comment by phue** [comment]
Interesting, but this will affect the next scikit-image release as well, regardless of the pillow breakage.

**Comment by hmaarrfk** [comment]
i totally agree, but that is a "tomorrow" problem ;)

**Comment by hmaarrfk** [comment]
Pillow 7.1.2 has just been released. 

I'm inclined to close this for today. What do you think?

**Comment by phue** [comment]
Absolutely, good that it is now resolved upstream

**Comment by hmaarrfk** [comment]
Nice. Yeah. Itis really nice when projects can respond to quick bugs like that

**Pull request closed without merging**